File: backend-java/src/main/java/com/empire/employeefinder/service/UserService.java
package com.empire.employeefinder.service;

import com.empire.employeefinder.dto.request.UserLoginRequestDto;
import com.empire.employeefinder.dto.request.UserRegisterRequestDto;
import com.empire.employeefinder.dto.request.UserUpdateRequestDto;
import com.empire.employeefinder.dto.response.UserLoginResponseDto;
import com.empire.employeefinder.dto.response.UserRegisterResponseDto;
import com.empire.employeefinder.dto.response.UserResponseDto;
import com.empire.employeefinder.model.enums.Role;

import java.util.List;

public interface UserService {

    UserRegisterResponseDto registration(UserRegisterRequestDto userRegisterRequestDto);

    UserLoginResponseDto authentication(UserLoginRequestDto userDto);

    UserResponseDto getById(Long userid);

    List<UserResponseDto> getAll(String searchField, String parameter, String sortField, String sortDirection, int pageSize, int pageNumber, String login);

    UserResponseDto changeUserRole(Long userId, Role newRole);

    void deleteById(Long userId);

    UserRegisterResponseDto changePassword(UserUpdateRequestDto userDto);
}

File: frontend-react/src/services/UserService.js
import axios from 'axios';

const USERS_URL = "http://localhost:8081/api/v1/users";

class UserService {

  getAllSortedUsersByPages = (sortField, sortDirection = "asc", pageSize, pageNumber) => {
    return axios.get(`${USERS_URL}?sortField=${sortField}&sortDirection=${sortDirection}&pageSize=${pageSize}&pageNumber=${pageNumber}`);
  };


  getAllSortedUsersSearched = (searchField, parameter, sortField, pageSize, pageNumber, sortDirection = "asc") => {
    return axios.get(`${USERS_URL}?searchField=${searchField}&parameter=${parameter}&sortField=${sortField}&sortDirection=${sortDirection}&pageSize=${pageSize}&pageNumber=${pageNumber}`);
  };

  updateUserRole = (userId, newRole) => {
    return axios.put(`${USERS_URL}/${userId}/change-role`, null, {
      params: { newRole }
    });
  }

  deleteUserById = (id) => {
    return axios.delete(`${USERS_URL}/${id}`);
  }
}

export default new UserService();